lib/class-wp-webfonts-provider-local.php

```diff
@@ -0,0 +1,263 @@
+	 * }
+	 * @font-face{
+	 *      font-family:"Source Serif Pro";
+	 *      font-style:italic;
+	 *      font-weight:200 900;
+	 *      font-stretch:normal;
+	 *      src:local("Source Serif Pro"), url('/assets/fonts/source-serif-pro/SourceSerif4Variable-Italic.ttf.woff2') format('woff2');
+	 * }
+	 * </code>
+	 *
+	 * @since 6.0.0
+	 *
+	 * @return string The `@font-face` CSS.
+	 */
+	public function get_css() {
+		$css = '';
+
+		foreach ( $this->webfonts as $webfont ) {
+			// Order the webfont's `src` items to optimize for browser support.
+			$webfont = $this->order_src( $webfont );
+
+			// Build the @font-face CSS for this webfont.
+			$css .= '@font-face{' . $this->build_font_face_css( $webfont ) . '}';
+		}
+
+		return $css;
+	}
+
+	/**
+	 * Order `src` items to optimize for browser support.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $webfont Webfont to process.
+	 * @return array
+	 */
+	private function order_src( array $webfont ) {
+		if ( ! is_array( $webfont['src'] ) ) {
+			$webfont['src'] = (array) $webfont['src'];
+		}
+
+		$src         = array();
+		$src_ordered = array();
+
+		foreach ( $webfont['src'] as $url ) {
+			// Add data URIs first.
+			if ( 0 === strpos( trim( $url ), 'data:' ) ) {
+				$src_ordered[] = array(
+					'url'    => $url,
+					'format' => 'data',
+				);
+				continue;
+			}
+			$format         = pathinfo( $url, PATHINFO_EXTENSION );
+			$src[ $format ] = $url;
+		}
+
+		// Add woff2.
+		if ( ! empty( $src['woff2'] ) ) {
+			$src_ordered[] = array(
+				'url'    => $src['woff2'],
+				'format' => 'woff2',
+			);
+		}
+
+		// Add woff.
+		if ( ! empty( $src['woff'] ) ) {
+			$src_ordered[] = array(
+				'url'    => $src['woff'],
+				'format' => 'woff',
+			);
+		}
+
+		// Add ttf.
+		if ( ! empty( $src['ttf'] ) ) {
+			$src_ordered[] = array(
+				'url'    => $src['ttf'],
+				'format' => 'truetype',
+			);
+		}
+
+		// Add eot.
+		if ( ! empty( $src['eot'] ) ) {
+			$src_ordered[] = array(
+				'url'    => $src['eot'],
+				'format' => 'embedded-opentype',
+			);
+		}
+
+		// Add otf.
+		if ( ! empty( $src['otf'] ) ) {
+			$src_ordered[] = array(
+				'url'    => $src['otf'],
+				'format' => 'opentype',
+			);
+		}
+		$webfont['src'] = $src_ordered;
+
+		return $webfont;
+	}
+
+	/**
+	 * Builds the font-family's CSS.
+	 *
+	 * @since 6.0.0
+	 *
+	 * @param array $webfont Webfont to process.
+	 * @return string This font-family's CSS.
+	 */
+	private function build_font_face_css( array $webfont ) {
+		$css = '';
+
+		// Wrap font-family in quotes if it contains spaces.
+		if (
+			false !== strpos( $webfont['font-family'], ' ' ) &&
+			false === strpos( $webfont['font-family'], '"' ) &&
+			false === strpos( $webfont['font-family'], "'" )
+		) {
+			$webfont['font-family'] = '"' . $webfont['font-family'] . '"';
+		}
+
+		foreach ( $webfont as $key => $value ) {
+
+			// Skip "provider".
```

Comment:
It's not immediately clear why the skip step is needed

Reply:
So that we don't serialize to css the stuff in the `provider` key. A possibly better approach was to add to the font definition a `props` key sibling to `provider` containing everything we serialize here, then no skip is needed.

Reply:
Added some extra info in the comment to explain why we skip this 👍